cocotb/share/makefiles/Makefile.sim

```diff
@@ -130,8 +130,6 @@ CUSTOM_SIM_DEPS += $(shell find $(COCOTB_PY_DIR)/cocotb/ -name "*.py")
 # This triggers a recompilation of the simulation if cocotb library is updated.
 CUSTOM_SIM_DEPS += $(shell find $(LIB_DIR)/ -name "*")
 
-PWD := $(shell pwd)
```

Comment:
Are we sure all the other makefiles define this? It was added recently to fix a big where we forgot to define it I think

Reply:
#1300

Reply:
It was used to set `PYTHONPATH` now is here: https://github.com/cocotb/cocotb/blob/master/cocotb/__init__.py#L63

Reply:
So to confirm, there are no makefiles left that use but do not define this variable?

Reply:
Yes. To what I can see. CI should fail also.

Reply:
Here's one that seems undefined: https://github.com/cocotb/cocotb/blob/master/tests/designs/avalon_module/Makefile#L42

Reply:
It is in CI so how does this work?

Reply:
I think that's the only one

Reply:
HA: https://unix.stackexchange.com/questions/174990/what-is-pwd-vs-current-working-directory

Reply:
And we do `SHELL := bash` also here: https://github.com/cocotb/cocotb/blob/master/cocotb/share/makefiles/Makefile.sim#L34

Reply:
Should I remove all this `$(shell pwd)` and just use `$(PWD)` ? Separate PR?

Reply:
I don't think we should do that, apparently the variable is empty within sudo

Reply:
Let's worry about that later, anyway

Reply:
But I think we should add the one missing call to that file, for consistency

Reply:
BTW: https://pubs.opengroup.org/onlinepubs/009604599/utilities/xcu_chap02.html 
Maybe should even avoid `PWD:=$(shell pwd)`. Yes. For another time I guess.

Reply:
`$(CURDIR)` is possibly a better choice, for a follow-up PR.

Reply:
>But I think we should add the one missing call to that file, for consistency

Should be done.